internal/controller/proxmoxcluster_controller.go

```diff
@@ -135,13 +159,6 @@ func (r *ProxmoxClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 }
 
 func (r *ProxmoxClusterReconciler) reconcileDelete(ctx context.Context, clusterScope *scope.ClusterScope) (reconcile.Result, error) {
-	// We want to prevent deletion unless the owning cluster was flagged for deletion.
-	if clusterScope.Cluster.DeletionTimestamp.IsZero() {
```

Comment:
Mmm... Can you elaborate on the use case? This only works when you put a finalizer on the owner cluster, because the owner cluster is needed for deletion. So, if we mark the owner cluster for deletion, but don't actually delete it yet, this can only achieved by a finalizer. Or, is placing an OwnerRef also prevents deletion?

Reply:
simply, the ProxmoxCluster can be deleted only if the root cluster is marked for deletion.

Reply:
Ok, reverted this back to what it was.